synthesizer/process/src/lib.rs

```diff
@@ -108,7 +139,7 @@ impl<N: Network> Process<N> {
         lap!(timer, "Synthesize credits program keys");
 
         // Add the 'credits.aleo' stack to the process.
-        process.add_stack(stack);
+        process.credits = Some(Arc::new(stack));
```

Comment:
would it make sense for `Stack::new` to already return an `Arc<Stack>`?

Reply:
Hmm I think I prefer the flexibility and more idiomatic return of Stack